project hackyourweather/hackyourweather/src/Components/SearchComponent/index.js

```diff
@@ -1,52 +1,50 @@
-import React, { useState, useEffect } from "react";
-import './styles.css'
-import CityBox from './../CityBox'
-
-
-
-const APIKEY = '4bde5fff659a96c59438421c777e914e'
-
+import React, { useState } from 'react';
+import './styles.css';
+import CityBox from './../CityBox';
 
+const APIKEY = '4bde5fff659a96c59438421c777e914e';
 
 function SearchComponent() {
+	let [ cityName, setCityName ] = useState('');
+	let [ cityData, setCityData ] = useState([]);
 
-    let [cityName, setCityName] = useState('')
-    let [cityData, setCityData] = useState()
-
-
+	const apiUrl = `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${APIKEY}`;
 
+	function submitHandler(e) {  
+		e.preventDefault();
 
+        fetch(apiUrl)
+            .then((response) => response.json())
+            .then((data) => {
+                if (data.cod !== '404') {
 
-    const apiUrl = `http://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${APIKEY}`
-
-    async function submitHandler(e) {
-        e.preventDefault()
-
-        await fetch(apiUrl)
-            .then(response => response.json())
-            .then(data => {
-                setCityData(data)
-                console.log('the setted data is ', data)
+                    setCityData([...cityData, data]);
+                }
+                else if (data.cod ==='404') { alert("City can't be found, please check your spelling") }
 
             })
 
-    }
+        setCityName('')
 
-    useEffect(() => {
+    }
 
-    }, [cityData])
+    function removeCity(props) {
 
+        if (cityData.length > 0) {
+            setCityData(cityData.filter(city => city.id !== props.id))
```

Comment:
Reduce the dependencies with the outside world!
```suggestion
            setCityData((prevCityData) => prevCityData.filter(city => city.id !== props.id))
```